lib/src/css_mirroring/bidi_css_gen.dart

```diff
@@ -0,0 +1,334 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+library scissors.src.css_mirroring.bidi_css_gen;
+
+import 'dart:async';
+
+import 'package:csslib/parser.dart' show parse;
+import 'package:csslib/visitor.dart'
+    show RuleSet, StyleSheet, TreeNode, Declaration, Directive, MediaDirective, HostDirective, PageDirective, CharsetDirective, FontFaceDirective, ImportDirective, NamespaceDirective;
+import 'package:quiver/check.dart';
+import 'package:source_maps/refactor.dart';
+import 'package:source_span/source_span.dart';
+
+import '../utils/enum_parser.dart';
+import 'transformer.dart' show CssMirroringSettings, Direction, flipDirection;
+
+enum RetentionMode {
+  keepBidiNeutral,  /// to keep parts of css which is direction independent eg: color and width
+  keepOriginalBidiSpecific, /// to keep direction dependent parts of original css eg: margin.
+  keepFlippedBidiSpecific /// to keep direction dependent parts of flipped css.
+}
+
+/// GenerateBidiCss generates a Css which comprises of Orientation Neutral, Orientation Specific and Flipped Orientation Specific parts.
```

Comment:
Good job on the comments!
I'm wondering if that can be moved to a .md file + linked from the top-level README.md, maybe as a follow-up commit :-)